src/Spaces/dss.jl

```diff
@@ -695,19 +695,110 @@ function weighted_dss!(
     data,
     space::ExtrudedFiniteDifferenceSpace{S, H},
     ghost_buffer = nothing,
-) where {S, H <: SpectralElementSpace2D}
+) where {S, T <: Topology2D, H <: SpectralElementSpace2D{T}}
     hspace = space.horizontal_space
     topology = hspace.topology
-    if isnothing(ghost_buffer)
-        ghost_buffer = create_ghost_buffer(data, topology)
-    end
     dss_2d!(
         topology,
         data,
-        ghost_buffer,
+        nothing,
         local_geometry_data(space),
-        ghost_geometry_data(space),
+        nothing,
         hspace.local_dss_weights,
-        hspace.ghost_dss_weights,
+        nothing,
+    )
+end
+
+function weighted_dss!(
+    data,
+    space::ExtrudedFiniteDifferenceSpace{S, H},
+    ghost_buffer = nothing,
+) where {S, T <: DistributedTopology2D, H <: SpectralElementSpace2D{T}}
+    weighted_dss_init_comms!(data, space, ghost_buffer)
+    weighted_dss_internal!(data, space, ghost_buffer)
+    weighted_dss_ghost!(data, space, ghost_buffer)
+end
+
+function weighted_dss_init_comms!(
```

Comment:
A simpler name? Also should have a docstring
```suggestion
function weighted_dss_start!(
```